src/main/java/com/budgetapp/repositories/IncomeRepository.java

```diff
@@ -3,5 +3,9 @@
 import com.budgetapp.entities.Income;
 import org.springframework.data.jpa.repository.JpaRepository;
 
+import java.util.List;
+import java.util.Optional;
+
 public interface IncomeRepository extends JpaRepository<Income, Long> {
+    Optional<List<Income>> findByUserId(final Long userId);
```

Comment:
This shouldn't be an optional

```suggestion
    List<Income> findAllByUserId(final Long userId);
```

Reply:
This is not resolved.